cmd/testoidcauth/main.go

```diff
@@ -0,0 +1,676 @@
+// Copyright (C) MongoDB, Inc. 2022-present.
```

Comment:
I made this an executable rather than using go test since this is what gssapi is doing... but also I know that when I go to test GCP and Azure that I'm going to want to build ahead of time, and I'm not sure if go test can build an executable (cargo test can).

Reply:
Also, here's hoping statically linking glibc works with go, since that was necessary to get the rust tests working :)

Reply:
`go test` can indeed build a binary. However, following the current convention seems fine. If/when we want to make these Go tests, we can convert them all at the same time.